testing/scenariobuilder/todoerr_scenario.go

```diff
@@ -62,10 +80,29 @@ func (s *TodoErrScenario) String() string {
 }
 
 type TodoErrForJSON struct {
-	Type     string `json:"type"`
-	Filename string `json:"filename"`
-	Line     int    `json:"line"`
-	Message  string `json:"message"`
+	Type     string            `json:"type"`
+	Filename string            `json:"filename"`
+	Line     int               `json:"line"`
+	Message  string            `json:"message"`
+	Metadata map[string]string `json:"metadata"`
+}
+
+func (s *TodoErrForJSON) isEqual(other *TodoErrForJSON) bool {
+	basic := s.Type == other.Type &&
+		s.Filename == other.Filename &&
+		s.Line == other.Line &&
+		s.Message == other.Message &&
+		len(s.Metadata) == len(other.Metadata)
+	if !basic {
+		return false
+	}
+	for k, v := range s.Metadata {
+		if other.Metadata[k] != v {
+			return false
+		}
+	}
```

Comment:
You can extract the metadata traversal in a separate func e.g. `metadatasMatch(this, other)` and then append it to the stream of checks at line 91. That way, you can directly return the stream of checks rather than assigning it to a variable & doing this processing afterwards.

Reply:
Extracted metadata checks to a `func metadatasMatch(this, other *TodoErrForJSON) bool`